File: examples/Scala/durasub.scala
/*
 * Durable subscriber
 *
 * @author Giovanni Ruggiero
 * @email giovanni.ruggiero@gmail.com
 *
 */

import org.zeromq.ZMQ

object durasub {
  def main(args : Array[String]) {

		val context = ZMQ.context(1)

		// Connect our subscriber socket
		val subscriber = context.socket(ZMQ.SUB)

		// Synchronize with the publisher
		val sync = context.socket(ZMQ.PUSH)

		subscriber.setIdentity("Hello".getBytes)
		subscriber.subscribe("".getBytes)
		subscriber.connect("tcp://localhost:5565")
		sync.connect("tcp://localhost:5564")
		sync.send("".getBytes, 0)

		// Get updates, expect random Ctrl-C death
		var msg = ""
	  do {
			msg = new String(subscriber.recv(0))
			println(msg)
		} while (!msg.equalsIgnoreCase("END"))
	}
}

File: examples/Java/durasub.java
import org.zeromq.ZMQ;

/**
 * Durable subscriber
 *
 * @author Faruk Akgul
 * @email faakgul@gmail.com
 */

public class durasub {
    public static void main(String[] args) {
        ZMQ.Context context = ZMQ.context(1);

        // Connect our subscriber socket
        ZMQ.Socket subscriber = context.socket(ZMQ.SUB);
        subscriber.setIdentity("hello".getBytes());

        // Synchronize with the publisher
        ZMQ.Socket sync = context.socket(ZMQ.PUSH);

        subscriber.subscribe("".getBytes());
        subscriber.connect("tcp://localhost:5565");
        sync.connect("tcp://localhost:5564");
        sync.send("".getBytes(), 0);

        // Get updates, expect random Ctrl-C death
        String msg = "";
        while (!msg.equalsIgnoreCase("END")) {
            msg = new String(subscriber.recv(0));
            System.out.println(msg);
        }
    }
}
